src/overlays/actors/ovl_En_Dy_Extra/z_en_dy_extra.c

```diff
@@ -21,15 +33,102 @@ const ActorInit En_Dy_Extra_InitVars = {
     (ActorFunc)EnDyExtra_Update,
     (ActorFunc)EnDyExtra_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dy_Extra/EnDyExtra_Init.s")
+void EnDyExtra_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnDyExtra_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDyExtra* this = THIS;
+
+    osSyncPrintf("\n\n");
+    //"Big fairy effect"
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 大妖精効果 ☆☆☆☆☆ %d\n" VT_RST, this->actor.params);
+    this->unk_150 = this->actor.params;
+    this->unk_15C.x = 0.025f;
+    this->unk_15C.y = 0.039f;
+    this->unk_15C.z = 0.025f;
+    this->unk_168 = this->actor.posRot.pos;
+    this->actor.gravity = -0.2f;
+    this->unk_158 = 1.0f;
+    this->unk_154 = 0x3C;
+    this->actionFunc = func_809FF7AC;
+}
+
+void func_809FF7AC(EnDyExtra* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxF(&this->actor.gravity, 0.0f, 0.1f, 0.005f);
+    if (this->actor.posRot.pos.y < -55.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+    if (this->unk_154 == 0) {
+        if (this->unk_152 != 0) {
+            this->unk_154 = 0xC8;
+            this->actionFunc = func_809FF840;
+        }
+    }
+}
+
+void func_809FF840(EnDyExtra* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxF(&this->actor.gravity, 0.0f, 0.1f, 0.005f);
+    if (this->unk_154 == 0 || this->unk_158 < 0.02f) {
+        Actor_Kill(&this->actor);
```

Comment:
with actor kill usually an early return makes more sense vs and else cause its like the condition to stop actor execution entirely, so everything just ends there
both are completely valid tho but just to keep the pattern consistent we can change it here